sentinel5dl/__init__.py

```diff
@@ -89,17 +90,41 @@ def __http_request(path, filename=None):
     logger.debug(f'Requesting {url}')
     with open(filename, 'wb') if filename else io.BytesIO() as f:
         curl = pycurl.Curl()
+        if ca_info:
+            curl.setopt(pycurl.CAINFO, ca_info)
         curl.setopt(curl.URL, url.encode('ascii', 'ignore'))
 
         curl.setopt(curl.USERPWD, f'{USER}:{PASS}')
         curl.setopt(curl.WRITEDATA, f)
         curl.setopt(curl.FAILONERROR, True)
 
+        # Use a Certificate Authority (CA) bundle if set
         if ca_info:
             curl.setopt(pycurl.CAINFO, ca_info)
 
-        curl.perform()
+        # abort if data transfer is slower than 30 bytes/sec during 60 seconds
+        curl.setopt(pycurl.LOW_SPEED_TIME, 60)
+        curl.setopt(pycurl.LOW_SPEED_LIMIT, 30)
+
+        # try to execute curl.perform() up to 10 times if it is not working
+        for retrys in range(10):
+            try:
+                curl.perform()
+
+            except pycurl.error as err:
+
+                # if last try failed too
+                if retrys >= 9:
+                    raise err
+
```

Comment:
When we are not raising an error, I would suggest logging the error:

```suggestion
logger.warning("Error performing the HTTP request, Retrying: %s", err)
```

Reply:
done